gbpservice/nfp/tests/unit/nfp/configurator/test_firewall.py

```diff
@@ -0,0 +1,887 @@
+        self._test_fw_event_creation('DELETE_FIREWALL')
+
+    def test_get_notifications_generic(self):
+        self._test_notifications()
+
+class FWaasRpcManagerTestCase(unittest.TestCase):
+    ''' Fwaas RPC receiver for Firewall module '''
+
+    def __init__(self, *args, **kwargs):
+        super(FWaasRpcManagerTestCase, self).__init__(*args, **kwargs)
+        self.fo = FakeObjects()
+
+    @mock.patch(__name__ + '.FakeObjects.sc')
+    @mock.patch(__name__ + '.FakeObjects.conf')
+    def _get_FWaasRpcManager_object(self, conf, sc):
+        agent = fw.FWaasRpcManager(sc, conf)
+        return agent, sc
+
+    def _test_event_creation(self, method):
+        agent, sc = self._get_FWaasRpcManager_object()
+        arg_dict = {'context': self.fo.context,
+                    'firewall': self.fo.firewall,
+                    'host': self.fo.host}
+        with mock.patch.object(sc, 'new_event', return_value='foo') as (
+                                                        mock_sc_event), \
+            mock.patch.object(sc, 'post_event') as mock_sc_rpc_event:
+            call_method = getattr(agent, method.lower())
+            call_method(self.fo.context, self.fo.firewall, self.fo.host)
+
+            mock_sc_event.assert_called_with(id=method, data=arg_dict)
+            mock_sc_rpc_event.assert_called_with('foo')
+
+    def test_create_firewall_fwaasrpcmanager(self):
+        ''' create_firewall method in RPC Receiver '''
+        self._test_event_creation('CREATE_FIREWALL')
+
+    def test_update_firewall_fwaasrpcmanager(self):
+        ''' update_firewall method in RPC Receiver '''
+        self._test_event_creation('UPDATE_FIREWALL')
+
+    def test_delete_firewall_fwaasrpcmanager(self):
+        ''' delete_firewall method in RPC Receiver '''
+        self._test_event_creation('DELETE_FIREWALL')
+
+
+class GenericConfigRpcManagerTestCase(unittest.TestCase):
+    ''' Generic Config RPC receiver for Firewall module '''
+
+    def __init__(self, *args, **kwargs):
+        super(GenericConfigRpcManagerTestCase, self).__init__(
+                                                        *args, **kwargs)
+        self.fo = FakeObjects()
+
+    @mock.patch(__name__ + '.FakeObjects.sc')
+    @mock.patch(__name__ + '.FakeObjects.conf')
+    def _get_GenericConfigRpcManager_object(self, conf, sc):
+        agent = gc.GenericConfigRpcManager(sc, conf)
+        return agent, sc
+
+    def _test_event_creation(self, method):
+        agent, sc = self._get_GenericConfigRpcManager_object()
+        arg_dict = {'context': self.fo.context,
+                    'kwargs': self.fo.kwargs}
+        with mock.patch.object(
+                    sc, 'new_event', return_value='foo') as mock_sc_event, \
+            mock.patch.object(sc, 'post_event') as mock_sc_rpc_event:
+            call_method = getattr(agent, method.lower())
+
+            call_method(self.fo.context, self.fo.kwargs)
+
+            if 'HEALTHMONITOR' in method:
```

Comment:
I abstracted this function with a single one now called _send_event(). Small changes needed for this test.